public/react/Configure/Teams/ConfigureTeams.tsx

```diff
@@ -0,0 +1,77 @@
+import type { FC } from "react";
+
+import type { NormalizedIssue, NormalizeIssueConfig } from "../../../jira/normalized/normalize";
+import type { SprintDefaults, UseSaveDefaultConfiguration, UseDefaultConfiguration } from "./services/team-defaults";
+
+import React from "react";
+import { useForm } from "react-hook-form";
+import Form, { FormHeader } from "@atlaskit/form";
+import { useQueryClient } from "@tanstack/react-query";
+
+import TextField from "./components/TextField";
+import { createNormalizeConfiguration } from "./shared/normalize";
+
+export interface ConfigureTeamsProps {
+  appKey: string;
+  useSave: UseSaveDefaultConfiguration;
+  useDefaults: UseDefaultConfiguration;
+  normalizedIssues?: Array<NormalizedIssue>;
+  onInitialDefaultsLoad?: (overrides: Partial<NormalizeIssueConfig>) => void;
+  onUpdate?: (overrides: Partial<NormalizeIssueConfig>) => void;
+}
+
+export type DefaultFormFields = SprintDefaults;
+
+export interface FieldUpdates<TProperty extends keyof DefaultFormFields> {
+  name: TProperty;
+  value: DefaultFormFields[TProperty];
+}
+
+const ConfigureTeams: FC<ConfigureTeamsProps> = ({ appKey, onUpdate, onInitialDefaultsLoad, useSave, useDefaults }) => {
+  const queryClient = useQueryClient();
+
+  const defaults = useDefaults({ appKey, onInitialDefaultsLoad });
+  const save = useSave({ appKey });
+
+  const { register, handleSubmit, getValues } = useForm<DefaultFormFields>({ defaultValues: defaults });
+
+  function update<TProperty extends keyof DefaultFormFields>({ name, value }: FieldUpdates<TProperty>) {
```

Comment:
nitpick: A cleanup that can be done is to wrap the logic of the `save` inside a function.
example:
```
function performSave(values: DefaultFormFields) {
    save(values, {
      onSuccess: () => {
        onUpdate?.(createNormalizeConfiguration(values));
        queryClient.invalidateQueries({
          queryKey: ["configuration", "default"],
        });
      },
    });
  }
  function update<TProperty extends keyof DefaultFormFields>({
    name,
    value,
  }: FieldUpdates<TProperty>) {
    const currentValues = getValues();
    const updatedValues = { ...currentValues, [name]: value };
    performSave(updatedValues);
  }

  return (
    <Form
      onSubmit={() =>
        handleSubmit((values) => {
          performSave(values);
        })
      }
    >
```

Reply:
Yeah good call. I'm not confident this won't change a bit still though. Mind if we pick this up in the next one?

Reply:
Yeah sure since its still not final